src/tools.ts

```diff
@@ -68,6 +68,56 @@ const scheduleTask = tool({
     return `Task scheduled for type "${when.type}" : ${input}`;
   },
 });
+
+/**
+ * Tool to list all scheduled tasks
+ * This executes automatically without requiring human confirmation
+ */
+const getScheduledTasks = tool({
+  description: "List all tasks that have been scheduled",
+  parameters: z.object({}),
+  execute: async () => {
+    const agent = agentContext.getStore();
+    if (!agent) {
+      throw new Error("No agent found");
+    }
+    try {
+      const tasks = agent.getSchedules();
+      if (!tasks || tasks.length === 0) {
+        return "No scheduled tasks found.";
+      }
+      return tasks.map((task) => `[${task.id}] ${task.payload}`).join("\n");
```

Comment:
should this return just the raw data instead?

Reply:
@threepointone updated!